src/utp_stream.cpp

```diff
@@ -206,6 +206,30 @@ void utp_socket_impl::release_packet(packet_ptr p)
 	m_sm.release_packet(std::move(p));
 }
 
+void utp_socket_impl::insert_packet(packet_ptr p)
+{
+	// If we're stalled we'll need to resend
+	if (m_stalled)
+	{
+		p->need_resend = true;
+		m_needs_resend.push_back(p.get());
+	}
+
+	packet_ptr old = m_outbuf.insert(m_seq_nr, std::move(p));
+	if (old)
+	{
+//		TORRENT_ASSERT(reinterpret_cast<utp_header*>(old->buf)->seq_nr == m_seq_nr);
```

Comment:
did you try un-commenting this? It looks like a reasonable thing to assert.

Reply:
Uncommented in f4fd64d, I just assumed there was a reason it was commented [9 years ago](https://github.com/arvidn/libtorrent/commit/103d574cb0d4cef4468586628bbfde42d3b6e66d#diff-a9b0002dfebb81b315f1d19d346361c385368345d731117eefad87b4b9817ca7).

Reply:
I was just curious to know if you had tested it at all.